simulator/scheduler/plugin/plugins.go

```diff
@@ -141,30 +141,14 @@ func NewPluginConfig(pc []configv1.PluginConfig) ([]configv1.PluginConfig, error
 	ret := make([]configv1.PluginConfig, 0, len(pluginConfig))
 	for name, arg := range pluginConfig {
 		// add plugin configs for default plugins.
-		ret = append(ret, configv1.PluginConfig{
-			Name: name,
-			Args: *arg,
-		})
-	}
-
-	defaultpls, err := config.RegisteredMultiPointPluginNames()
-	if err != nil {
-		return nil, xerrors.Errorf("get default score/filter plugins: %w", err)
-	}
-
-	for _, name := range defaultpls {
-		pc, ok := pluginConfig[name]
-		if !ok {
-			continue
-		}
-
-		ret = append(ret, configv1.PluginConfig{
-			Name: pluginName(name),
-			Args: *pc,
-		})
-
-		// avoid adding same plugin config.
-		delete(pluginConfig, name)
```

Comment:
Looks like we should keep these lines?
If there's no plugin config given from users, then we still should use `defaultpls`.

Reply:
I removed these lines because they would essentially add a "wrapped" version of the plugin config, but only for the in-tree and out-of-tree plugins, as we don't know about the WASM ones at this stage yet.

By adding a "wrapped" version of the plugin config to all of them, the logic is simplified, and it ensures that wrapped WASM plugins will be able to find their config.

Reply:
> as we don't know about the WASM ones at this stage yet.

So.. what do you mean by this? I'm not sure how the original version didn't work out for the wasm extension plugin configuration.

Reply:
`NewPluginConfig` is called by `ConvertConfigurationForSimulator` which happens before we have a `k8s.io/kubernetes/pkg/scheduler/apis/config.KubeSchedulerConfiguration`, and we need it to be able to know which plugins are WASM plugins.

I could change the place where we add the WASM plugins to the registry to be in `NewConfigs` before the call to `ConvertConfigurationForSimulator`. That would remove the need for the changes above (which didn't sit well with me in the first place). However I'm not sure this is conceptually the right place for this either!

Reply:
Alright, I have reworked it to be able to register the wasm plugins during configuration read (`debuggablescheduler.NewConfigs`). That simplifies the implementation and decouples create the wasm plugin registry from using it.